src/System.Management.Automation/engine/CommandCompletion/ExtensibleCompletion.cs

```diff
@@ -102,7 +102,127 @@ IEnumerable<CompletionResult> CompleteArgument(
             string parameterName,
             string wordToComplete,
             CommandAst commandAst,
-            IDictionary fakeBoundParameters);
+            IDictionary fakeBoundParameters)
+        {
+            CommandName = commandName;
+            ParameterName = parameterName;
+            WordToComplete = wordToComplete;
+            CommandAst = commandAst;
+            FakeBoundParameters = fakeBoundParameters;
+            return ShouldComplete ? GetMatchingResults(wordToComplete) : [];
+        }
+
+        /// <summary>
+        /// Gets the name of the command that needs argument completion.
+        /// </summary>
+        protected static string? CommandName { get; private set; }
```

Comment:
Why static? I think it is unique per instance.

Reply:
Thinking more I believe all these members initially come from CompleteArgument() - they are dynamic. So it makes no sense to keep them. It is unnecessary limitation.